### PR TITLE
feat(core): add jenkins artifact type

### PR DESCRIPTION
**Pull request by Jammy-Louie**
spinnaker/spinnaker#4101

### CODE REVIEW

File: igor-web/src/main/groovy/com/netflix/spinnaker/igor/jenkins/JenkinsBuildMonitor.groovy [hunk 1]
```diff
@@ -227,7 +225,13 @@ class JenkinsBuildMonitor extends CommonPollingMonitor<JobDelta, JobPollingDelta
             registry.counter(missedNotificationId.withTag("monitor", getClass().simpleName)).increment()
             return
         }
-        echoService.get().postEvent(new BuildEvent(content: new BuildContent(project: project, master: master)))
+        BuildEvent buildEvent = new BuildEvent(content: new BuildContent(project: project, master: master))
+        buildEvent.content.project.lastBuild.artifacts.each {
```
Comment (jkschneider) [inline igor-web/src/main/groovy/com/netflix/spinnaker/igor/jenkins/JenkinsBuildMonitor.groovy]: Can `artifacts` be null here?

File: igor-web/src/main/groovy/com/netflix/spinnaker/igor/jenkins/client/JenkinsClient.groovy [hunk 2]
```diff
@@ -41,7 +41,7 @@ interface JenkinsClient {
     @GET('/api/xml?tree=jobs[name,jobs[name,jobs[name,jobs[name,jobs[name,jobs[name,jobs[name,jobs[name,jobs[name,jobs[name]]]]]]]]]]')
     JobList getJobs()
 
-    @GET('/job/{jobName}/api/xml?exclude=/*/build/action[not(totalCount)]&tree=builds[number,url,duration,timestamp,result,building,url,fullDisplayName,actions[failCount,skipCount,totalCount]]')
+    @GET('/job/{jobName}/api/xml?exclude=/*/build/action[not(totalCount)]&tree=builds[number,url,duration,timestamp,result,building,url,fullDisplayName,artifacts[displayPath,fileName,relativePath],actions[failCount,skipCount,totalCount]]')
```
Comment (jkschneider) [inline igor-web/src/main/groovy/com/netflix/spinnaker/igor/jenkins/client/JenkinsClient.groovy]: Why exclude these? Aren't these fields what the artifact reference is ultimately derived from?

File: igor-web/src/main/groovy/com/netflix/spinnaker/igor/jenkins/client/model/BuildArtifact.java [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2014 Netflix, Inc.
+ * Copyright 2019 Pivotal, Inc.
```
Comment (jkschneider) [inline igor-web/src/main/groovy/com/netflix/spinnaker/igor/jenkins/client/model/BuildArtifact.java]: Do not change license headers on updates.

File: igor-web/src/main/groovy/com/netflix/spinnaker/igor/jenkins/JenkinsBuildMonitor.groovy (hunk 1 above)
Comment (Jammy-Louie) [reply igor-web/src/main/groovy/com/netflix/spinnaker/igor/jenkins/JenkinsBuildMonitor.groovy]: sure, empty list not sufficient?

Comment (Jammy-Louie) [reply igor-web/src/main/groovy/com/netflix/spinnaker/igor/jenkins/JenkinsBuildMonitor.groovy]: 👍

File: igor-web/src/main/groovy/com/netflix/spinnaker/igor/jenkins/client/model/BuildArtifact.java (hunk 3 above)
Comment (Jammy-Louie) [reply igor-web/src/main/groovy/com/netflix/spinnaker/igor/jenkins/client/model/BuildArtifact.java]: 👍

File: igor-web/src/main/groovy/com/netflix/spinnaker/igor/jenkins/client/JenkinsClient.groovy (hunk 2 above)
Comment (Jammy-Louie) [reply igor-web/src/main/groovy/com/netflix/spinnaker/igor/jenkins/client/JenkinsClient.groovy]: @eleftherias and I noticed that if we didn't add that as part of the exclude we wouldn't get the details to derive the reference

**Pull request merged**